Replicator/tests/ReplicatorAPITest.cc

```diff
@@ -613,6 +613,215 @@ TEST_CASE_METHOD(ReplicatorAPITest, "Is Document Pending", "[C][Push]") {
+    params.socketFactory = _socketFactory;
+
+    int expectedPending = 0;
+    SECTION("Normal") {
+        expectedPending = 100;
+    }
+
+    SECTION("Filtered") {
+        expectedPending = 99;
+        params.replCollection.callbackContext = this;
+        params.pushFilter = []( C4CollectionSpec collectionSpec, C4String docID, C4String revID,
+                                C4RevisionFlags flags, FLDict flbody, void *context) {
+            auto test = (ReplicatorAPITest*)context;
+            c4repl_getStatus(test->_repl);
+            return FLSlice_Compare(docID, "0000005"_sl) != 0;
+        };
+    }
+
+    SECTION("Set Doc IDs") {
+        expectedPending = 2;
+        FLEncoder e = FLEncoder_New();
+        FLEncoder_BeginDict(e, 1);
+        FLEncoder_WriteKey(e, FLSTR(kC4ReplicatorOptionDocIDs));
+        FLEncoder_BeginArray(e, 2);
+        FLEncoder_WriteString(e, FLSTR("0000002"));
+        FLEncoder_WriteString(e, FLSTR("0000004"));
+        FLEncoder_EndArray(e);
+        FLEncoder_EndDict(e);
+        options = FLEncoder_Finish(e, nullptr);
+        params.replCollection.optionsDictFleece = C4Slice(options);
+        FLEncoder_Free(e);
+    }
+
+    _repl = c4repl_newLocal(db, (C4Database*)db2, params, ERROR_INFO(err));
+    REQUIRE(_repl);
+
+    FLSliceResult_Release(options);
+
+    C4SliceResult encodedDocIDs = c4repl_getPendingDocIDs(_repl, Eisley, ERROR_INFO(err));
+    REQUIRE(encodedDocIDs != nullslice);
+    FLArray docIDs = FLValue_AsArray(FLValue_FromData(C4Slice(encodedDocIDs), kFLTrusted));
+    CHECK(FLArray_Count(docIDs) == expectedPending);
+    c4slice_free(encodedDocIDs);
+
+    c4repl_start(_repl, false);
+    REQUIRE_BEFORE(5s, c4repl_getStatus(_repl).level == kC4Stopped);
+    encodedDocIDs = c4repl_getPendingDocIDs(_repl, Eisley, &err);
+    CHECK(err.code == 0);
+    CHECK(encodedDocIDs == nullslice);
+
+}
+#endif
+
+#ifdef COUCHBASE_ENTERPRISE
+TEST_CASE_METHOD(ReplicatorAPITest, "Pending Document IDs Non-Existent Collection", "[C][Push]") {
+    ExpectingExceptions x;
+    // Create collection in the db and import documents
+    C4CollectionSpec Kyber = {"kyber"_sl, "crystal"_sl};
+    auto collKyber = createCollection(db, Kyber);
+    importJSONLines(sFixturesDir + "names_100.json", collKyber);
+
+    // Create the collection in the database to be replicated to
+    createDB2();
+    createCollection(db2, Kyber);
+
+    // Create collection spec but do not create collection in db
+    C4CollectionSpec Republic = {"republic"_sl, "galactic"_sl};
+
+    // Set options for the replication
+    C4Error err;
+    repl::C4ReplParamsCollection params{Kyber};
+    params.push = kC4OneShot;
+    params.pull = kC4Disabled;
+    params.callbackContext = this;
+    params.socketFactory = _socketFactory;
+
+    _repl = c4repl_newLocal(db, (C4Database*)db2, params, ERROR_INFO(err));
+    REQUIRE(_repl);
+
+    C4SliceResult encodedDocIDs = c4repl_getPendingDocIDs(_repl, Republic, ERROR_INFO(err));
```

Comment:
ERROR_INFO not really needed here.  Its purpose is to log the error that occurred if the next check fails, but the next check is checking the error code itself anyway.